### PR TITLE
Allow to pass arguments to fileserver.update runner

### PR DESCRIPTION
**Pull request by sathieu**
### What does this PR do?

Allow to pass arguments to fileserver.update runner

This is useful to only update one GitFS remote.

### What issues does this PR fix or reference?

None. But #53622 is related.

### Previous Behavior

No way to only update one GitFS repo.

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

**Comment by sathieu** [comment]
@twangboy @waynew @terminalmage Please review :eyeglasses:

**Comment by sathieu** [comment]
@dwoz I've added tests (and rebased).

**Comment by waynew** [comment]
@sathieu thanks for the PR and sorry for the delay in getting this reviewed! I've merged the most recent master changes in. Next week we're going to be pretty busy with some planning, but I've written this down to make sure we get this looked at by 2nd week in March (2 weeks from now)

**Review by cmcmarrow — APPROVED** [review]

**Review by cmcmarrow — CHANGES_REQUESTED** [review]
Can we also get test for the changes you made in gitfs?

**Comment by waynew** [comment]
@sathieu I brought this up with the team, and `**kwargs` is usually an anti-pattern in Salt. Unless there's a *really* good reason, you should specify the keyword args that you want to be able to use.

With `**` it bypasses our args checker, so we can no longer check that states are being used correctly.

If you want to avoid having to provide an argument, default arguments *are* OK!

Let me know if you have any questions!

File: salt/fileserver/__init__.py [hunk 1]
```diff
@@ -492,17 +492,17 @@ def clear_lock(self, back=None, remote=None):
                 errors.extend(bad)
         return cleared, errors
 
-    def update(self, back=None):
+    def update(self, back=None, **kwargs):
```
Comment (waynew) [inline salt/fileserver/__init__.py:495]: Please update this signature to include the specific arguments that should be passed

File: salt/runners/fileserver.py [hunk 2]
```diff
@@ -322,7 +322,7 @@ def empty_dir_list(saltenv="base", backend=None):
     return fileserver.file_list_emptydirs(load=load)
 
 
-def update(backend=None):
+def update(backend=None, **kwargs):
```
Comment (waynew) [inline salt/runners/fileserver.py:325]: Please update this signature to include the specific arguments that should be passed

**Review by waynew — CHANGES_REQUESTED** [review]
Could you please update these to no longer take `**kwargs`?

File: salt/fileserver/__init__.py (hunk 1 above)
Comment (sathieu) [reply salt/fileserver/__init__.py:495]: The arguments are backend specific. How to handle this?

Another solution would be to pass this `remotes` arg to all backend, but it doesn't always makes sense (for example the `roots` backend).

Currently logic throws an error when the parameter is incorrect. Isn't this sufficient?

Comment (sathieu) [reply salt/fileserver/__init__.py:482]: I've added a test for this.

**Comment by sathieu** [comment]
I've added a test to ensure that invalid parameters throw an error.

Remaining todo:
> Can we also get test for the changes you made in gitfs?

**Comment by sathieu** [comment]
@waynew @dwoz @cmcmarrow @twangboy Please review.

**Review by waynew — DISMISSED** [review]
That seems reasonable to me - I'll bring it up to the rest of the team.

**Review by cmcmarrow — DISMISSED** [review]

**Comment by cmcmarrow** [comment]
@sabaini thanks for adding tests. This should get merged shortly after sodium gets released.

**Comment by sathieu** [comment]
@cmcmarrow I've added the tests 😄

**Comment by sathieu** [comment]
@cmcmarrow You've marked this PR *Merge Ready*. I've now rebased and fixed `pre-commit` and `Pylint`. Hope you merge it now ...

**Comment by sathieu** [comment]
@sagetherage You've removed the *Merge Ready* tag here. What can I do help merging this PR (and others #56316, #57288)?

**Comment by sagetherage** [comment]
we removed that label all together from the project as it doesn't mean what it says - this needs to be rebased with master, and needs a completed review.

**Comment by sathieu** [comment]
@sagetherage @cmcmarrow @waynew  @twangboy Any chance to get this reviewed and merged soon? It was marked merge-ready few months ago...

Thanks

**Review by twangboy — DISMISSED** [review]

File: salt/fileserver/__init__.py (hunk 1 above)
Comment (s0undt3ch) [reply salt/fileserver/__init__.py:480]: Can we be more specific on which kwargs to accept?
If not, why not.

Comment (s0undt3ch) [reply salt/fileserver/__init__.py:480]: Guess I'm late to the discussion...

Comment (waynew) [reply salt/fileserver/__init__.py:480]: Not merged yet :trollface:

**Comment by sagetherage** [comment]
The team is working on getting the test suite to green overall and then merging will start, again. I haven't updated the branch since the branch builds are failing, so I will get back to this soon.

**Comment by sathieu** [comment]
@twangboy @sagetherage Hello. Sorry to insist, but can this be merged in magnesium ?

This MR exists since Oct 2019.

NB: I've merged master and resolved the conflict.

**Review by dwoz — APPROVED** [review]

**Pull request merged**